mono/mini/mini-llvm.c

```diff
@@ -1010,6 +1010,18 @@ simd_op_to_llvm_type (int opcode)
 #endif
 }
 
+static void
+set_preserveall_cc (LLVMValueRef func)
+{
+	/*
+	 * xcode10 doesn't seem to support preserveall on watchos, it fails with:
+	 * fatal error: error in backend: Unsupported calling convention
+	 */
+#ifndef TARGET_WATCHOS
```

Comment:
Can we use runtime-if instead of preprocessor-if? Some day it'd be nice to have one multi-targeting binary. Every pound-if is a step away from that.